### DISCUSSION TITLE
Dashboards are not rendering after upgrade Superset from version 2.1.0 to version 3.0.1 on a Kubernetes cluster

### DISCUSSION THREAD

**Discussion by vinod-av** [apache/superset#26318] [Q&A / Help]
Hi Team,

I am attempting to upgrade Superset from version 2.1.0 to version 3.0.1 on a Kubernetes cluster connected to a PostgreSQL database. We haven't made any custom code changes. While I successfully deployed Superset 3.0.1 and executed the 'superset upgrade db' command for database migration, none of the old charts are rendering. 

This is the screenshot of charts before upgrading 

![screenshot of charts before upgrading](https://github.com/apache/superset/assets/1920998/6f2016c6-fd41-4686-8994-6fa7fbf555b1)

Screenshot of charts after upgrading 
![after_upgrading_with_browser_logs](https://github.com/apache/superset/assets/1920998/200fdeca-32c4-4a61-b0fd-9c445017e9e0)

Below is the logs of init container which shows the database migration logs

`Upgrading DB schema...
logging was configured successfully
2023-12-20 12:47:17,662:INFO:superset.utils.logging_configurator:logging was configured successfully
2023-12-20 12:47:17,671:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>
/usr/local/lib/python3.9/site-packages/flask_limiter/extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade f3c2d8ec8595 -> 9c2a5681ddfd, convert key-value entries to json
INFO  [alembic.runtime.migration] Running upgrade 9c2a5681ddfd -> c0a3ea245b61, remove_show_native_filters
INFO  [alembic.runtime.migration] Running upgrade c0a3ea245b61 -> d0ac08bb5b83, invert_horizontal_bar_chart_order
INFO  [alembic.runtime.migration] Running upgrade d0ac08bb5b83 -> b5ea9d343307, bar_chart_stack_options
INFO  [alembic.runtime.migration] Running upgrade b5ea9d343307 -> 07f9a902af1b, drop postgres enum constrains for tags
INFO  [alembic.runtime.migration] Running upgrade 07f9a902af1b -> 7e67aecbf3f1, chart-ds-constraint
INFO  [alembic.runtime.migration] Running upgrade 7e67aecbf3f1 -> 4ea966691069, cross-filter-global-scoping
INFO  [alembic.runtime.migration] Running upgrade 4ea966691069 -> 9ba2ce3086e5, migrate-pivot-table-v1-to-v2
INFO  [alembic.runtime.migration] Running upgrade 9ba2ce3086e5 -> 4c5da39be729, migrate_treemap_chart
INFO  [alembic.runtime.migration] Running upgrade 4c5da39be729 -> ae58e1e58e5c, migrate_dual_line_to_mixed_chart
INFO  [alembic.runtime.migration] Running upgrade ae58e1e58e5c -> 83e1abbe777f, drop access_request
INFO  [alembic.runtime.migration] Running upgrade 83e1abbe777f -> 90139bf715e4, add_currency_column_to_metrics
INFO  [alembic.runtime.migration] Running upgrade 90139bf715e4 -> 6fbe660cac39, add on delete cascade for tables references
INFO  [alembic.runtime.migration] Running upgrade 6fbe660cac39 -> 8e5b0fb85b9a, Add custom size columns to report schedule
INFO  [alembic.runtime.migration] Running upgrade 8e5b0fb85b9a -> 240d23c7f86f, update_tag_model_w_description
INFO  [alembic.runtime.migration] Running upgrade 240d23c7f86f -> f92a3124dd66, drop rouge constraints and tables
INFO  [alembic.runtime.migration] Running upgrade f92a3124dd66 -> 6d05b0a70c89, add on delete cascade for owners references
INFO  [alembic.runtime.migration] Running upgrade 6d05b0a70c89 -> 863adcf72773, delete obsolete Druid NoSQL slice parameters
INFO  [alembic.runtime.migration] Running upgrade 863adcf72773 -> a23c6f8b1280, cleanup erroneous parent filter IDs
INFO  [alembic.runtime.migration] Running upgrade a23c6f8b1280 -> bf646a0c1501, json_metadata
INFO  [alembic.runtime.migration] Running upgrade bf646a0c1501 -> 0769ef90fddd, Fix schema perm for datasets
INFO  [alembic.runtime.migration] Running upgrade 0769ef90fddd -> 2e826adca42c, Fix schema for log
INFO  [alembic.runtime.migration] Running upgrade 2e826adca42c -> 8ace289026f3, add on delete cascade for dashboard_slices
INFO  [alembic.runtime.migration] Running upgrade 8ace289026f3 -> 4448fa6deeb1, add on delete cascade for embedded_dashboards
INFO  [alembic.runtime.migration] Running upgrade 4448fa6deeb1 -> 9f4a086c2676, add_normalize_columns_to_sqla_model
INFO  [alembic.runtime.migration] Running upgrade 9f4a086c2676 -> ec54aca4c8a2, Increase ab_user.email field size
INFO  [alembic.runtime.migration] Running upgrade ec54aca4c8a2 -> 4b85906e5b91, add on delete cascade for dashboard_roles
Loaded your LOCAL configuration at [/app/pythonpath/superset_config.py]
Initializing roles...
logging was configured successfully
2023-12-20 12:47:37,051:INFO:superset.utils.logging_configurator:logging was configured successfully
2023-12-20 12:47:37,074:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>
/usr/local/lib/python3.9/site-packages/flask_limiter/extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
Syncing role definition
2023-12-20 12:47:48,209:INFO:superset.security.manager:Syncing role definition
Syncing Admin perms
2023-12-20 12:47:48,296:INFO:superset.security.manager:Syncing Admin perms
Syncing Alpha perms
2023-12-20 12:47:49,913:INFO:superset.security.manager:Syncing Alpha perms
Syncing Gamma perms
2023-12-20 12:47:51,197:INFO:superset.security.manager:Syncing Gamma perms
Syncing sql_lab perms
2023-12-20 12:47:52,431:INFO:superset.security.manager:Syncing sql_lab perms
Fetching a set of all perms to lookup which ones are missing
2023-12-20 12:47:53,928:INFO:superset.security.manager:Fetching a set of all perms to lookup which ones are missing
Creating missing datasource permissions.
2023-12-20 12:47:54,980:INFO:superset.security.manager:Creating missing datasource permissions.
Creating missing database permissions.
2023-12-20 12:47:55,089:INFO:superset.security.manager:Creating missing database permissions.
Cleaning faulty perms
2023-12-20 12:47:55,097:INFO:superset.security.manager:Cleaning faulty perms
Loaded your LOCAL configuration at [/app/pythonpath/superset_config.py]
Creating admin user...
logging was configured successfully
2023-12-20 12:48:03,593:INFO:superset.utils.logging_configurator:logging was configured successfully
2023-12-20 12:48:03,640:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>
/usr/local/lib/python3.9/site-packages/flask_limiter/extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
Loaded your LOCAL configuration at [/app/pythonpath/superset_config.py]
Recognized Database Authentications.
Error! User already exists admin`

Please let me know if you require additional details, and I will provide them.

**Comment by michael-s-molina** [comment]
@vinod-av You added the same screenshot twice. Could you update the screenshot and also attach the browser console output?

**Reply by vinod-av** [reply]
@michael-s-molina Oops sorry, I have corrected it.

**Comment by michael-s-molina** [comment]
@vinod-av I can't see all the browser logs because it needs scrolling. Can you confirm there are no errors (not warnings) in the `Network` and `Console` tabs?

Taking the `Weight Band` chart as an example, are the queries the same between version 2.1.0 and 3.0.1? You can check the query by clicking on the 3 dots and `View query`.

**Reply by vinod-av** [reply]
@michael-s-molina I have checked again and can confirm that there are no errors in browser network.
Verified the queries, there is no changes in the query as well.

![image](https://github.com/apache/superset/assets/1920998/15203547-94a9-4ea4-bd42-ac7fbb192bdb)

**Reply by michael-s-molina** [reply]
When you execute the queries on SQL Lab, do you get any result or they return empty?
